appyx-interactions/common/src/commonMain/kotlin/com/bumble/appyx/transitionmodel/spotlight/interpolator/SpotlightSlider.kt

```diff
@@ -117,8 +119,27 @@ class SpotlightSlider<NavTarget : Any>(
             }
         }
 
-        // TODO fix with displacement is ready
-        override fun isVisible(): Boolean = true
+        override fun isVisible(): Boolean {
+            val leftEdgeOffsetDp = offset.displacedValue.value.x.value.roundToInt()
+            val rightEdgeOffsetDp = leftEdgeOffsetDp + containerWidth.value.roundToInt()
+            val visibleWindowLeftEdge = calculatedWindowLeftEdge(clipToBounds)
+            val visibleWindowRightEdge = calculateWindowRightEdge()
+            return (rightEdgeOffsetDp <= visibleWindowLeftEdge || leftEdgeOffsetDp >= visibleWindowRightEdge).not()
+        }
```

Comment:
Can we move this into properties? See the branch `visibility-per-field`

Reply:
yes, will change that in a separate PR